IO/Ex1.cpp

```diff
@@ -0,0 +1,108 @@
+
+#include <stdexcept> //runtime_error
+#include <cassert>
+
+#include "Ex1.hpp"
+
+/************************************* Main Functions *******************************************/
+
+//Ctor
+FileHandler::FileHandler(const std::string path, std::ios_base::openmode permissions)
+: m_path(path), 
+  m_permissions(permissions)
+{
+
+    m_file.open(path, permissions);
+    if (!m_file.is_open()) {
+        throw std::runtime_error("Failed to open file: " + path);
+    }
+}
+
+//CCtor
+FileHandler::FileHandler(FileHandler &other) 
+: m_path(other.m_path), 
+  m_permissions(other.m_permissions) {
+
+    m_file.open(m_path, m_permissions);
+    if (!m_file.is_open()) {
+        throw std::runtime_error("Failed to open file: " + m_path);
+    }   
+
+    auto g = other.m_file.tellg();
+    auto p = other.m_file.tellp();
+    if (g == -1 || p == -1) {
+        throw std::runtime_error("Failed to do tellg/p on file: " + other.m_path);
+    }
+}
+
+//MCtor
+FileHandler::FileHandler(FileHandler &&other) noexcept 
+: m_file(std::move(other.m_file)),
+  m_path(std::move(other.m_path)),
+  m_permissions(other.m_permissions)
+{
+    other.m_permissions = std::ios_base::openmode(0);
+}
+
+//Copy Assignment
+FileHandler& FileHandler::operator=(FileHandler &other) {
```

Comment:
please add a comment about why this `operator=` receives non const reference

Reply:
added in the header

Reply:
nevermind . missed the comment in the header